tests/spec/urlparser-spec.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * Test for URL Parameter parsing
+ */
+
+
+let defaultUrl = "https://python.microbit.org/v/1.1"; 
+describe("Testing different URL formats", () => {
+    test("Check if no Parameter is recognised", () =>{
+        expect(global.urlparse("parameter", defaultUrl) === undefined).toBeTruthy();
```

Comment:
You can use `expect(global.urlparse("parameter", defaultUrl).toBeUndefined()`.
And the same with all the other checks, have a look at all the options from https://jestjs.io/docs/en/expect.html.

Reply:
Updated all undefined result tests to use .toBeUndefined()

Reply:
Used .toBeNull() with all the null returning tests as well